pkg/cli/admin/catalog/mirror.go

```diff
@@ -375,29 +375,76 @@ func WriteManifests(out io.Writer, name, dir, icspScope string, mapping map[stri
 		return err
 	}
 
-	icsp, err := generateICSP(out, name, icspScope, mapping)
+	defer func() {
+		fmt.Fprintf(out, "wrote mirroring manifests to %s\n", dir)
+	}()
+
+	if dest.Type != imagesource.DestinationRegistry {
+		fmt.Fprintf(out, "mirroring to %s, skipping ICSP and CatalogSource creation\n", dest.Type)
+		return nil
+	}
+
+	icsp, err := generateICSP(out, source.Ref.Name, icspScope, mapping)
 	if err != nil {
 		return err
 	}
 
 	if err := ioutil.WriteFile(filepath.Join(dir, "imageContentSourcePolicy.yaml"), icsp, os.ModePerm); err != nil {
 		return fmt.Errorf("error writing ImageContentSourcePolicy")
 	}
-	fmt.Fprintf(out, "wrote mirroring manifests to %s\n", dir)
+
+	catalogSource, err := generateCatalogSource(source, mapping)
+	if err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(filepath.Join(dir, "catalogSource.yaml"), catalogSource, os.ModePerm); err != nil {
+		return fmt.Errorf("error writing CatalogSource")
+	}
```

Comment:
Should these files be uniquely named - icsp.yaml, catalogSource.yaml, mapping.txt? (I see "imageContentSourcePolicy.yaml" is not part of this PR)  See how the signature file is named here: https://github.com/openshift/oc/blob/master/pkg/cli/admin/release/mirror.go#L302-#L313  and a PR to add the writing of icsp file w/ `oc adm release mirror` will also use this pattern to name the icsp file.  Won't these files be overwritten when mirror cmd is run more than once, and is this a potential problem?

Reply:
the manifests all go into a directory, so I copied what must-gather does and added a timestamp to the containing folder

Reply:
sg, and like you said below, we can align `oc adm catalog mirror` with `oc adm release mirror` in a follow-up PR if necessary.